iceoryx_hoofs/test/moduletests/test_cxx_string.cpp

```diff
@@ -1089,153 +1090,634 @@ TYPED_TEST(stringTyped_test, CompareOperatorGreaterEqResultFalseWithDifferentCap
+    ASSERT_THAT(sut.unsafe_assign(temp), Eq(true));
+
+    char testChar = 'M';
+    EXPECT_THAT(sut == testChar, Eq(false));
+    EXPECT_THAT(testChar == sut, Eq(false));
+    EXPECT_THAT(sut != testChar, Eq(true));
+    EXPECT_THAT(testChar != sut, Eq(true));
 }
 
-TYPED_TEST(stringTyped_test, CompareOperatorFixedStringNotEqualSTDStringWithSameSizeResultTrue)
+/// @note bool operator<(const char& rhs) const noexcept
+/// bool operator<=(const char& rhs) const noexcept
+/// bool operator>(const char& rhs) const noexcept
+/// bool operator>=(const char& rhs) const noexcept
+TYPED_TEST(stringTyped_test, CompareOperatorsWithDifferentChar)
 {
-    ::testing::Test::RecordProperty("TEST_ID", "99a40704-9d0b-49f9-bd5c-4c87b9f2ba00");
+    ::testing::Test::RecordProperty("TEST_ID", "c818b150-b926-4f3c-8405-6327303f12f6");
     using MyString = typename TestFixture::stringType;
     constexpr auto STRINGCAP = MyString::capacity();
-    string<STRINGCAP> testFixedString("M");
-    std::string testSTDString = "L";
-    EXPECT_THAT(testFixedString != testSTDString, Eq(true));
+    string<STRINGCAP> sut1("M");
+    string<STRINGCAP> sut2("F");
+
+    char testChar = 'L';
+    EXPECT_THAT(sut1 < testChar, Eq(false));
+    EXPECT_THAT(sut1 <= testChar, Eq(false));
+    EXPECT_THAT(sut2 < testChar, Eq(true));
+    EXPECT_THAT(sut2 <= testChar, Eq(true));
+    EXPECT_THAT(sut1 > testChar, Eq(true));
+    EXPECT_THAT(sut1 >= testChar, Eq(true));
+    EXPECT_THAT(sut2 > testChar, Eq(false));
+    EXPECT_THAT(sut2 >= testChar, Eq(false));
+
+    EXPECT_THAT(testChar < sut1, Eq(true));
+    EXPECT_THAT(testChar <= sut1, Eq(true));
+    EXPECT_THAT(testChar < sut2, Eq(false));
+    EXPECT_THAT(testChar <= sut2, Eq(false));
+    EXPECT_THAT(testChar > sut1, Eq(false));
+    EXPECT_THAT(testChar >= sut1, Eq(false));
+    EXPECT_THAT(testChar > sut2, Eq(true));
+    EXPECT_THAT(testChar >= sut2, Eq(true));
+}
+
+TYPED_TEST(stringTyped_test, CompareOperatorsWithEqualChar)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "6d6344a8-d8d6-426d-b288-b429eb9a6eca");
+    this->testSubject = "M";
+
+    char testChar = 'M';
+    EXPECT_THAT(this->testSubject < testChar, Eq(false));
+    EXPECT_THAT(this->testSubject <= testChar, Eq(true));
+    EXPECT_THAT(this->testSubject > testChar, Eq(false));
+    EXPECT_THAT(this->testSubject >= testChar, Eq(true));
+
+    EXPECT_THAT(testChar < this->testSubject, Eq(false));
+    EXPECT_THAT(testChar <= this->testSubject, Eq(true));
+    EXPECT_THAT(testChar > this->testSubject, Eq(false));
+    EXPECT_THAT(testChar >= this->testSubject, Eq(true));
+}
+
+TYPED_TEST(stringTyped_test, CompareOperatorsWithDifferentCharWithDifferentSize)
+{
+    ::testing::Test::RecordProperty("TEST_ID", "c9f2e472-3fbf-4e08-a0ee-6bb725e7c0d7");
+    using MyString = typename TestFixture::stringType;
+    constexpr auto STRINGCAP = MyString::capacity();
+
+    std::string temp1(STRINGCAP + 5U, 'M');
+    string<STRINGCAP + 5U> sut1;
```

Comment:
Could we rename the sut into better understandable names. Since we compare them always with `testChar` I think the name `sutSmallerTestChar`, `sutEqualTestChar` and `sutGreaterTestChar` would help in understanding the single conditions you verify below.

And maybe we can create 3 `EXPECT_THAT` groups for smaller, equal and greater